shim/src/ctrl/setsockopt.c

```diff
@@ -58,5 +59,5 @@ int __setsockopt(int sockfd, int level, int optname, const void *optval, socklen
         WARN("%s is not supported", "TCP_ULP");
     }
 
-    return (demi_setsockopt(sockfd, level, optname, optval, optlen));
+    return (__demi_setsockopt(sockfd, level, optname, optval, optlen));
```

Comment:
Shouldn't we do the same as we do in `getpeername()`? That isCheck if return value is different from zero. If it is, we should set errno and return `-1`.